make_count/make_count.tex

```diff
@@ -0,0 +1,293 @@
+
+Moreover, for some input vector \texttt{v} drawn from \texttt{input\_domain}, $\function$ either returns $\texttt{exact\_int\_cast(len(data), TO)}$, which will be of type \texttt{TO} by the definition of \texttt{exact\_int\_cast}; or, if the casting fails, it returns $\texttt{get\_max\_consecutive\_int(TO)}$ which, from our definition of $\texttt{get\_max\_consecutive\_int}$, will be of type \texttt{TO}. Therefore, since our output domain is always some value of type \texttt{TO}, we see that $\function$ has the appropriate output domain \texttt{output\_domain}.
+\end{proof}
+
+\question{Is the second paragraph in the proof above of ``Appropriate output domain'' necessary?}
+
+\begin{proof} \textbf{(Part 2 -- domain-metric compatibility).}
+
+The \texttt{input\_domain} is \texttt{VectorDomain(AllDomain(TIA))}. Because our \texttt{input\_metric} of \texttt{SymmetricDistance} is compatible with any domain of the form \\\texttt{VectorDomain(inner\_domain)}, and because \texttt{VectorDomain(AllDomain(TIA))} is of this form, we see that it is compatible with our \texttt{input\_metric} of \texttt{SymmetricDistance}.
+
+The \texttt{output\_domain} is \texttt{AllDomain(TO)}. Because our \texttt{output\_metric} of \texttt{SymmetricDistance} is compatible with any domain of the form \texttt{AllDomain(T)} where \texttt{T} has the trait \texttt{Sub(Output=T)}, and because \texttt{AllDomain(TO)} is of this form and has the necessary trait, we see that it is compatible with our \texttt{output\_metric} of \texttt{AbsoluteDistance}.
+\end{proof}
+
+\begin{proof} \textbf{(Part 3 -- stability relation).} We consider two inputs: a vector $\texttt{u}$ of elements of type \texttt{TIA}; and a vector $\texttt{v}$ of elements of type \texttt{TIA}. (This \texttt{input\_domain} is specified in the pseudocode in section \ref{sec:python-pseudocode}.)
+
+Assume it is the case that $\texttt{stability\_relation}(\din, \dout) = \True$. From  the stability relation provided on line \ref{line:stability-relation}, this means that $\texttt{inf\_cast}(\din, \texttt{TO}) \leq \dout$. Recall that \texttt{inf\_cast} will cast $\din$ to a value at least as large as $\din$, so this assumption that $\texttt{stability\_relation}$ is $\True$ also means that $\din \leq \dout$. Also assume that \texttt{v}, \texttt{w} are $\din$-close under the symmetric distance metric (in accordance with the \texttt{input\_metric} specified in the preconditions in section \ref{sec:python-pseudocode}).
+
+We now refer to the definition of symmetric distance provided in the \href{https://www.overleaf.com/project/60d214e390b337703d200982}{Proof Definitions} document; the definition is copied here for convenience:
+
+\begin{definition}[Symmetric distance]
+\label{defn:symm-dist}
+Let $u,v$ be vectors of elements drawn from domain $\mathcal{X}$. Define $m_v(\ell)$ as the multiplicity of element $\ell$ in vector $v$. For example, if $v$ contains five instances of the number ``21'', then $m_v(21) = 5$.

```

Comment:
are we using m_l or h_l as our notation?

Reply:
We will use h_l.

I created a new definition in the proof definitions doc for a generic symmetric distance and generic histogram that can apply to any (ordered or unordered) list. The updated proof definitions doc will hopefully be pushed within the next two days.